.pre-commit-config.yaml

```diff
@@ -38,29 +39,22 @@ repos:
       - id: numpydoc-validation
         files: ^pyvista/
 
-  - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v1.11.2
+  - repo: https://github.com/pyvista/pre-commit-mypy
+    rev: v0.1dev6
```

Comment:
A `v0.1` release will be need to be created once the review is complete.

The main thing this fork does is:

1. Remove the "mirror" part. The repo's version is no longer sync'd with the `mypy` version. Instead, we control the `mypy` version via `additional_dependencies`, which is controlled here in the `pyvista` repo instead.
2. Remove the [default args](https://github.com/pre-commit/mirrors-mypy/blob/main/.pre-commit-hooks.yaml#L7): `["--ignore-missing-imports", "--scripts-are-modules"]` so that we have full control over what is passed to `mypy`
4. Use `pass_filenames: false`, as recommended by the `mypy` devs here: https://github.com/python/mypy/issues/13916 (see https://github.com/pyvista/pyvista/pull/6738#issuecomment-2439471798 for context)